src/WebJobs.Script/Diagnostics/FileLoggerProvider.cs

```diff
@@ -17,7 +19,20 @@ public FileLoggerProvider(ScriptHostConfiguration config, Func<string, LogLevel,
             _filter = filter;
         }
 
-        public ILogger CreateLogger(string categoryName) => new FileLogger(categoryName, _config, _filter);
+        public ILogger CreateLogger(string categoryName)
+        {
+            switch (categoryName)
+            {
+                case ScriptConstants.LogCategoryHostNodeConsoleLogs:
```

Comment:
I think we can probably generalize and consolidate the logic so we pass flags indicating we want to isolate the logs based on category. That would eliminate one type and make this logic here more flexible (and not specific to Node)